pilot/pkg/model/push_context.go

```diff
@@ -1707,12 +1707,21 @@ func (ps *PushContext) BestEffortInferServiceMTLSMode(service *Service, port *Po
 		return MTLSUnknown
 	}
 
-	// First , check mTLS settings from beta policy (i.e PeerAuthentication) at namespace / mesh level.
+	// 1. Check service instances' tls mode, mainly used for headless service.
+	if service.Resolution != ClientSideLB {
+		instances := ps.InstancesByPort(service, port.Port, nil)
+		// Assume all instances are same.
```

Comment:
Isn't a primary goal of auto mtls that we don't make this assumption? Seems like if an instance scales down now suddenly mtls switches which is not quite right.

Reply:
For headless service, there is no reason to be different.

>Seems like if an instance scales down now suddenly mtls switches which is not quite right.

Not quite understand this

Reply:
Agree with @howardjohn . My original thought in the issue's thread is: if we can have one outbound cluster per pod ip, we can do it correctly, but at least for now the headless service is still one cluster per service?

If you already get all `instances`, we can check all their tls label here and only all has TLSMode label, then infer it as MtlsPermissive? Just to be more consistent.

Reply:
like, 

```go
for _, i := range instances {
  if i.TLSMode == DisableTlsModeLabel { return MTLSDisable }
}
```

Reply:
sgtm, I have not strong feelings about this.